diff -ru FreeRDP-1.1.0-beta1.orig/libfreerdp/locale/keyboard_xkbfile.c FreeRDP-1.1.0-beta1/libfreerdp/locale/keyboard_xkbfile.c
--- FreeRDP-1.1.0-beta1.orig/libfreerdp/locale/keyboard_xkbfile.c	2013-03-13 04:59:04.000000000 +0900
+++ FreeRDP-1.1.0-beta1/libfreerdp/locale/keyboard_xkbfile.c	2013-08-14 22:54:02.649489920 +0900
@@ -55,7 +55,7 @@
 	{ "RTRN",	RDP_SCANCODE_RETURN}, // not KP
 	{ "LFSH",	RDP_SCANCODE_LSHIFT},
 	{ "LALT",	RDP_SCANCODE_LMENU},
-	{ "CAPS",	RDP_SCANCODE_CAPSLOCK},
+	{ "CAPS",	RDP_SCANCODE_LCONTROL},
 	{ "ESC",	RDP_SCANCODE_ESCAPE},
 	{ "SPCE",	RDP_SCANCODE_SPACE},
 	{ "AE10",	RDP_SCANCODE_KEY_0},
